File: lcs.java
package Java;
import java.util.*;

public class lcs {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the first String");
        String a = sc.next();
        System.out.println("Enter the second String");
        String b = sc.next();
        int c[][] = new int[a.length()+1][b.length()+1];
        for (int i = 0; i <= a.length(); i++) {
            for (int j = 0; j <= b.length(); j++) {
                if (i == 0 || j == 0) {
                    c[i][j] = 0;
                }
                else if (a.charAt(i - 1) == b.charAt(j - 1)) {
                    c[i][j] = c[i - 1][j - 1] + 1;
                }
                else {
                    c[i][j] = Math.max(c[i - 1][j], c[i][j - 1]);
                }
            }
        }
        for (int i = 0; i < c.length; i++) {
            System.out.println(Arrays.toString(c[i]));
        }

        sc.close();
    }
}

File: lcs.c
#include <stdio.h>
#include <string.h>

int main(int argc, char const *argv[])
{
    char a[100], b[100];    
    printf("Enter the first string ");
    scanf("%s", &a);
    printf("Enter the second string ");
    scanf("%s", &b);
    printf("\n");
    int c[100][100];
    for (int i = 0; i <= strlen(a); i++) {
        for (int j = 0; j <= strlen(b); j++) {
            if (i == 0 || j == 0) {
                c[i][j] = 0;
            }
            else if (a[i - 1] == b[j - 1]) {
                c[i][j] = c[i - 1][j - 1] + 1;
            }
            else {
                c[i][j] = (c[i - 1][j] < c[i][j - 1]) ? c[i][j - 1] : c[i - 1][j];
            }
        }
    }
    for (int i = 0; i <= strlen(a); i++)
    {
        for (int j = 0; j <= strlen(b); j++)
        {
            printf("%d ", c[i][j]);
        }
        printf("\n");
    }

    return 0;
}
